markups/asciidoc.py

```diff
@@ -0,0 +1,105 @@
+# vim: ts=4:sts=4:sw=4:expandtab
+
+# This file is part of python-markups module
+# License: 3-clause BSD, see LICENSE file
+# Copyright: (C) Dmitry Shachnev, 2012-2021
+
+import importlib
+from io import StringIO
+import markups.common as common
+from markups.abstract import AbstractMarkup, ConvertedMarkup
+
+
+class AsciidocMarkup(AbstractMarkup):
+    """Markup class for Asciidoc language.
+    Inherits :class:`~markups.abstract.AbstractMarkup`.
+
+    :param settings_overrides: optional dictionary of overrides for the
+                               `Docutils settings`_
+    :type settings_overrides: dict
+    """
+    name = 'asciidoc'
+    attributes = {
+        common.LANGUAGE_HOME_PAGE: 'https://asciidoc3.org',
+        common.MODULE_HOME_PAGE: 'https://asciidoc3.org/userguide.html',
+        common.SYNTAX_DOCUMENTATION: 'https://asciidoc3.org/userguide.html'
+    }
+
+    file_extensions = ('.adoc', '.asciidoc')
+    default_extension = '.adoc'
+
+    @staticmethod
+    def available():
+        try:
+            importlib.import_module('asciidoc3')
+            importlib.import_module('lxml')
+        except ImportError:
+            return False
+        return True
+
+    def __init__(self, filename=None):
+        AbstractMarkup.__init__(self, filename)
+        from asciidoc3 import asciidoc3api
+        self.converter_class = asciidoc3api.AsciiDoc3API
+
+    def convert(self, text):
+        import asciidoc3
+        from asciidoc3 import asciidoc3api
+        from lxml import etree
+        infile = StringIO()
+        outfile = StringIO()
+        infile.write(text)
+        infile.seek(0)
+        converter = self.converter_class(
```

Comment:
You are importing `asciidoc3api` in this function anyway, so you can remove `self.converter_class`.

```suggestion
        converter = asciidoc3api.AsciiDoc3API(
```